python/lsst/daf/butler/_butler.py

```diff
@@ -2080,12 +2088,16 @@ def import_(
             else:
                 _, format = os.path.splitext(filename)  # type: ignore
         elif filename is None:
-            filename = f"export.{format}"
-        if isinstance(filename, str) and directory is not None and not os.path.exists(filename):
-            filename = os.path.join(directory, filename)
+            filename = ResourcePath(f"export.{format}", forceAbsolute=False)
+        if directory is not None:
+            directory = ResourcePath(directory, forceDirectory=True)
+        if isinstance(filename, (str, ResourcePath, pathlib.Path, urllib.parse.ParseResult)):
```

Comment:
I guess this is the same as `ResourcePathExpression`, what happens if `ResourcePathExpression` definition changes?

Reply:
Yes. I was worrying about that. I couldn't initially think of a way to avoid the instance check and appease mypy. Unfortunately I can't say `isinstance(filename, ResourcePathExpression)`.

Reply:
One option is to add a class method to ResourcePath along the lines of ResourcePath.can_accept.  What do you think?

Reply:
Would it work if you define:

    ResourcePathExpression = str | urllib.parse.ParseResult | ResourcePath | Path

(after ResourcePath class is declared)? You can then say `if isinstance(filename, ResourcePathExpression)`.

Reply:
If Andy's last post works, I'm all for it.  I keep forgetting that `isinstance` now does unions.

Reply:
A quick test indicates that moving it to the end (so that we no longer need to quote the ResourcePath entry) does indeed work.